docs/proposals/pod-security-context.md

```diff
@@ -0,0 +1,468 @@
+    Capabilities *Capabilities `json:"capabilities,omitempty"`
+
+    // Run the container in privileged mode
+    Privileged *bool `json:"privileged,omitempty"`
+
+    // SELinuxOptions are the labels to be applied to the container
+    // and volumes
+    SELinuxOptions *SELinuxOptions `json:"seLinuxOptions,omitempty"`
+
+    // RunAsUser is the UID to run the entrypoint of the container process.
+    RunAsUser *int64 `json:"runAsUser,omitempty"`
+
+    // RunAsNonRoot indicates that the container should be run as a non-root user.  If the RunAsUser
+    // field is not explicitly set then the kubelet may check the image for a specified user or
+    // perform defaulting to specify a user.
+    RunAsNonRoot bool
+}
+
+// SELinuxOptions are the labels to be applied to the container.
+type SELinuxOptions struct {
+    // SELinux user label
+    User string `json:"user,omitempty"`
+
+    // SELinux role label
+    Role string `json:"role,omitempty"`
+
+    // SELinux type label
+    Type string `json:"type,omitempty"`
+
+    // SELinux level label.
+    Level string `json:"level,omitempty"`
+}
+```
+
+The new V1 API should be backward compatible with the existing API.  Backward compatibility is
+defined as:
+
+> 1.  Any API call (e.g. a structure POSTed to a REST endpoint) that worked before your change must
+>     work the same after your change.
+> 2.  Any API call that uses your change must not cause problems (e.g. crash or degrade behavior) when
+>     issued against servers that do not include your change.
+> 3.  It must be possible to round-trip your change (convert to different API versions and back) with
+>     no loss of information.
+
+Let's break down the fields of the `PodSecurityContext` and examine the backward compatibility rules
+for each class of field:
+
+1.  Fields that are purely additive have no backward compatibility concerns
+2.  For fields that are moving from the `PodSpec` to the `PodSecurityContext`:
+    1.  If the new field is set and the deprecated field is unset, the deprecated field is defaulted
+        to the value of the new field
+    2.  If the deprecated field is set and the new field is unset, the new field is defaulted to the
+        value of the deprecated field
+    3.  If the deprecated field and new field are both set, their values must match
+2.  For fields in `pod.Spec.SecurityContext.ContainerDefaults`:
+    1.  If the field **is set** in `pod.Spec.SecurityContext.ContainerDefaults`, for each container:
+        1.  If the field is not set in `container.SecurityContext`, the `container.SecurityContext`
+        field is set to the value of the `pod.Spec.SecurityContext.ContainerDefaults` field
+    2.  If the field **is not set** in `pod.Spec.SecurityContext.ContainerDefaults`, for each
+        container:
+        1.  If the field is not set in `container.SecurityContext`,  no field values are altered
+        2.  If the field is set in the `container.SecurityContext` of all containers and has a
```

Comment:
@bgrant0607 with the intention that the pod-level supplemental group can't be overriden by containers.  It will be a field directly on `PodSecurityContext`.  I want to add `GID` and supplemental group control to containers as well, but those are orthogonal to the shared pod-level supplemental group.